addons/stock/stock.py

```diff
@@ -1314,7 +1314,7 @@ def _create_link_for_product(operation_id, product_id, qty):
 
                         #check if the quant is matching the operation details
                         if ops.package_id:
-                            flag = quant.package_id and bool(package_obj.search(cr, uid, [('id', 'child_of', [ops.package_id.id])], context=context)) or False
+                            flag = quant.package_id == ops.package_id and bool(package_obj.search(cr, uid, [('id', 'child_of', [ops.package_id.id])], context=context)) or False
```

Comment:
hm, simpler is flag = quant.package_id == ops.package_id (all the rest can be left out, only perf loss)